File: LeetCode/Arrays/867-Transpose-Matrix.js
/**
 * Problem: 867. Transpose Matrix
 * Approach: Create Transposed Matrix
 *
 * The element at matrix[i][j] moves to answer[j][i].
 *
 * If the original matrix is m x n,
 * the transposed matrix will be n x m.
 *
 * Time Complexity: O(m * n)
 * Space Complexity: O(m * n)
 */

/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */

var transpose = function(matrix) {
    const m = matrix.length;
    const n = matrix[0].length;

    const ans = Array.from(
        { length: n },
        () => Array(m)
    );

    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            ans[j][i] = matrix[i][j];
        }
    }

    return ans;
};

File: LeetCode/Arrays/867-Transpose-Matrix.py
"""
Problem: 867. Transpose Matrix
Approach: Create Transposed Matrix

The element at matrix[i][j] moves to answer[j][i].

If the original matrix is m x n,
the transposed matrix will be n x m.

Time Complexity: O(m * n)
Space Complexity: O(m * n)
"""


class Solution:
    def transpose(self, matrix: list[list[int]]) -> list[list[int]]:
        m = len(matrix)
        n = len(matrix[0])

        ans = [[0] * m for _ in range(n)]

        for i in range(m):
            for j in range(n):
                ans[j][i] = matrix[i][j]

        return ans